tools/scripts/update-bumpenvs-yaml.py

```diff
@@ -200,15 +204,11 @@ def update_tag_for_image_type(subconf: Dict[str, str], new_tag: str) -> bool:
     builds = get_all_builds(commit)
     artifacts = get_all_artifacts(builds)
 
-    new_tags = {
-        **yaml.safe_load(artifacts["publish-tf1-cpu"]),
-        **yaml.safe_load(artifacts["publish-tf2-cpu"]),
-        **yaml.safe_load(artifacts["publish-tf1-gpu"]),
-        **yaml.safe_load(artifacts["publish-tf2-gpu"]),
-        **yaml.safe_load(artifacts["publish-tf25-gpu"]),
-        **yaml.safe_load(artifacts["publish-tf26-gpu"]),
-        **parse_packer_log(artifacts["packer-log"]),
-    }
+    tag_list = [
+        parse_packer_log(artifacts[artifact]) for artifact in PACKER_ARTIFACTS
+    ] + [yaml.safe_load(artifacts[artifact]) for artifact in DOCKER_ARTIFACTS]
+
+    new_tags = {k: d[k] for d in tag_list for k in d}
```

Comment:
Without really digging into the functions that generated this (which I'm about to do) it's not clear what the types or underlying meanings are here. I suspect we can pick much more descriptive names for k and d that would make this much more readable.

Reply:
this is a merge of all dicts in tag_list.
we had them written out by hand, and then you can do {**dict1, **dict2, **dict3}, but you can't do this when you have a list of them.

Reply:
A comment or separate function for this would be helpful. Also, it should use `.items()`.